src/http/server.rs

```diff
@@ -1596,6 +1608,12 @@ pub mod ws {
                 ..Default::default()
             };
 
+            let subproto_c_str; // SAFETY: same scope as httpd_register_uri_handler required!
```

Comment:
Nit: can you move this `let` binding inside the `if`? I don't see a reason why it should be outside.

Reply:
No - I tried that first, and it definitely triggers adverse behavior. My explanation: the variable gets dropped at the end of the `if` scope  -and the (raw) pointer becomes dangling before the (unsafe) call into `httpd_ws_recv_frame`

See also: https://doc.rust-lang.org/std/ffi/struct.CStr.html#method.as_ptr  (big WARNING)